src/components/SignUpForm/index.js

```diff
@@ -4,47 +4,83 @@ import "what-input";
 
 import "./SignUpForm.scss";
 
+import ENV from 'config/Environment';
+import MailchimpSubscribe from 'react-mailchimp-subscribe'
+
 class SignUpForm extends PureComponent {
   constructor(props) {
     super(...arguments);
 
     this.state = {
-      value: ""
+      emailAddress: '',
+      message: '',
     };
   }
 
   handleChange = event => {
-    this.setState({ value: event.target.value });
-  };
-
-  handleSubmit = event => {
-    event.preventDefault();
+    this.setState({ emailAddress: event.target.value });
   };
 
   render() {
+    const url = ENV.MAILCHIMP_URL;
+    const body = document.querySelector('body');
```

Comment:
I get why you did it this way, but the react way to do it is to use the ```onKeyUp``` prop, much like you did with with the onClick.

Reply:
I fixed this by using the `form` element with an `onSubmit` and changing the `input` to a `button` with a `type="submit"` seemed like the most accessible solution, is that alright?